cfg/conf.sample.php

```diff
@@ -70,13 +70,16 @@
 ; Check the documentation at https://content-security-policy.com/
 ; Note: If you use a bootstrap theme, you can remove the allow-popups from the sandbox restrictions.
 ; By default this disallows to load images from third-party servers, e.g. when they are embedded in pastes. If you wish to allow that, you can adjust the policy here. See https://github.com/PrivateBin/PrivateBin/wiki/FAQ#why-does-not-it-load-embedded-images for details.
-; cspheader = "default-src 'none'; manifest-src 'self'; connect-src *; script-src 'self'; style-src 'self'; font-src 'self'; img-src 'self' data:; media-src data:; object-src data:; Referrer-Policy: 'no-referrer'; sandbox allow-same-origin allow-scripts allow-forms allow-popups allow-modals"
+; cspheader = "default-src 'none'; manifest-src 'self'; connect-src *; script-src 'self' 'unsafe-eval'; style-src 'self'; font-src 'self'; img-src 'self' data: blob:; media-src blob:; object-src blob:; Referrer-Policy: 'no-referrer'; sandbox allow-same-origin allow-scripts allow-forms allow-popups allow-modals"
 
 ; stay compatible with PrivateBin Alpha 0.19, less secure
 ; if enabled will use base64.js version 1.7 instead of 2.1.9 and sha1 instead of
 ; sha256 in HMAC for the deletion token
 zerobincompatibility = false
 
+; enable or disable the warning message when the site is served over HTTP instead of HTTPS, defaults to true
```

Comment:
```suggestion
; enable or disable the warning message when the site is served over an insecure connection (insecure HTTP instead of HTTPS), defaults to true. Secure transport methods like Tor and I2P domains are automatically whitelisted.
; It is **strongly discouraged** recommend to disable this.
```

* clarify it whitelists things already
* clarify it is never encouraged to disable this

Reply:
I hadn't seen any such whitelist feature based on transport beyond HTTP? How do you detect the site is accessed via Tor, there will be no trace in the URL? If this is bothering anyone, the admin can just disable the warning, this is why its configurable.

It is enabled by default, so the setting is likely ignored until it becomes an issue. I assume that a person disabling the feature has a good reason for doing so, which we can't fathom. I suspect most folks will see it in local development environments.

Reply:
>  How do you detect the site is accessed via Tor, there will be no trace in the URL?

https://github.com/PrivateBin/PrivateBin/pull/407/files/42c20032204faf494cdb6e2d3cf99675b66c96ed#diff-cf28a0bc390eee2482509aca8212f831R4594

accessed via Tor != Tor hidden service (= what I called "Tor domain")

> I suspect most folks will see it in local development environments.

localhost is already whitelisted.

And suspections won't help. People often too easily disable security features. That's why I'd strongly argue to mention this in the config. (After all I've argued for not having any setting at al before, so this is already a compromise!)

Reply:
Look, I'll get this feature out of the door, because you requested it and said you have no time to do so. If you have time to discuss the exact wording of the message, please just change it, I don't mind that, but I do mind the time we spend on discussing such details.

Reply:
You can apply patches with one click on GitHub. Come on… use that tools. I basically changed it… :roll_eyes:

Reply:
I thought you had insisted on me signing every commit using GPG? Hence I have to use git CLI and can't commit in my IDE as it doesn't support GPG signing? Have I restricted myself in vain?

Reply:
Well… yeah, this was too easy. IMHO in feature branches (pull requests) we do not have to sign commits.

Important is that the merge commit is then signed, i.e. you review the code, merge it and sign it, so that you also sign that the code has been reviewed.

After all, if other contributors contribute code, it's also not git signed. 
AFAIK all git commit verifications will work. You usually verify the latest commit, so if that is signed it is all okay. (also `git pull --verify-signatures` works if the last merge commit is signed AFAIK.)